spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinRestTemplateSender.java

```diff
@@ -16,153 +16,71 @@
-	}
-
-	@Override
-	public Call<Void> sendSpans(List<byte[]> encodedSpans) {
-		if (this.closed) {
-			throw new ClosedSenderException();
-		}
-		return new HttpCall(this.endpoint, BytesMessageEncoder.JSON.encode(encodedSpans), this.restTemplate);
-	}
-
-	@Override
-	public CheckResult check() {
-		try {
-			sendSpans(List.of()).execute();
-			return CheckResult.OK;
-		}
-		catch (IOException | RuntimeException ex) {
-			return CheckResult.failed(ex);
-		}
-	}
-
-	@Override
-	public void close() throws IOException {
-		this.closed = true;
-	}
-
-	private static class HttpCall extends Call.Base<Void> {
-
-		/**
-		 * Only use gzip compression on data which is bigger than this in bytes.
-		 */
-		private static final DataSize COMPRESSION_THRESHOLD = DataSize.ofKilobytes(1);
+	private static class RestTemplateHttpPostCall extends HttpPostCall {
 
 		private final String endpoint;
 
-		private final byte[] body;
-
 		private final RestTemplate restTemplate;
 
-		HttpCall(String endpoint, byte[] body, RestTemplate restTemplate) {
+		RestTemplateHttpPostCall(String endpoint, byte[] body, RestTemplate restTemplate) {
+			super(body);
 			this.endpoint = endpoint;
-			this.body = body;
 			this.restTemplate = restTemplate;
 		}
 
 		@Override
-		protected Void doExecute() throws IOException {
-			HttpHeaders headers = new HttpHeaders();
-			headers.set("b3", "0");
-			headers.set("Content-Type", "application/json");
-			byte[] body;
-			if (needsCompression(this.body)) {
-				headers.set("Content-Encoding", "gzip");
-				body = compress(this.body);
-			}
-			else {
-				body = this.body;
-			}
-			HttpEntity<byte[]> request = new HttpEntity<>(body, headers);
-			this.restTemplate.exchange(this.endpoint, HttpMethod.POST, request, Void.class);
-			return null;
-		}
-
-		private boolean needsCompression(byte[] body) {
-			return body.length > COMPRESSION_THRESHOLD.toBytes();
+		public Call<Void> clone() {
+			return new RestTemplateHttpPostCall(this.endpoint, getBody(false), this.restTemplate);
 		}
 
 		@Override
-		protected void doEnqueue(Callback<Void> callback) {
-			try {
-				doExecute();
-				callback.onSuccess(null);
-			}
-			catch (IOException | RuntimeException ex) {
-				callback.onError(ex);
-			}
+		protected Void doExecute() {
+			HttpEntity<byte[]> request = new HttpEntity<>(getBody(), getDefaultHeaders());
+			this.restTemplate.exchange(this.endpoint, HttpMethod.POST, request, Void.class);
+			return null;
 		}
 
 		@Override
-		public Call<Void> clone() {
-			return new HttpCall(this.endpoint, this.body, this.restTemplate);
-		}
-
-		private byte[] compress(byte[] input) throws IOException {
-			ByteArrayOutputStream result = new ByteArrayOutputStream();
-			try (GZIPOutputStream gzip = new GZIPOutputStream(result)) {
-				gzip.write(input);
-			}
-			return result.toByteArray();
+		protected void doEnqueue(Callback<Void> callback) {
+			CompletableFuture.supplyAsync(this::doExecute).whenComplete((__, throwable) -> {
```

Comment:
We don't need the `CompletableFuture` in this case. The UrlConnectionSender implements the `enqueue` call without additional threading, so we should be fine here without introducing multi-threading.